src/util/system.cpp

```diff
@@ -965,6 +970,11 @@ bool ArgsManager::ReadConfigStream(std::istream& stream, const std::string& file
     return true;
 }
 
+fs::path ArgsManager::GetConfigFilePath()
+{
+    return GetConfigFile(GetPathArg("-conf", BITCOIN_CONF_FILENAME));
```

Comment:
The behavior has been slightly changed:

- on master:
```
$ ./src/qt/bitcoin-qt -regtest -conf=nothing
Error: Error reading configuration file: specified config file "nothing" could not be opened.

```

- this PR (6207b73e508d6cfc7583483d00261c66b0c348c3):
```
$ ./src/qt/bitcoin-qt -regtest -conf=nothing
Error: Error reading configuration file: specified config file "/home/hebasto/.bitcoin/nothing" could not be opened.

```

If it is intended, maybe document it?

Reply:
Right that's a good catch. We still _behave_ the same as previously, either fetching `conf` from an absolute path or completing the path with default datadir, but now because I simplified [reading the conf into a stream](https://github.com/bitcoin/bitcoin/pull/27073/commits/6207b73e508d6cfc7583483d00261c66b0c348c3#diff-19427b0dd1a791adc728c82e88f267751ba4f1c751e19262cac03cccd2822216L993), we return the full path we were trying in the error.

I agree that therefore there is a slight behaviour change, and personally I think this is more desirable than previously, where users were not told the absolute path the program was searching on for the config I would prefer to keep it this way, but not sure what documentation you would suggest adding in this case?

Reply:
> ... not sure what documentation you would suggest adding in this case?

The PR description? A commit message?

Reply:
I'm curious now, why functional tests did not catch this change?

Reply:
The reason is that we do not test relative `-conf=` arguments [here](https://github.com/bitcoin/bitcoin/blob/73966f75f67fb797163f0a766292a79d4b2c1b70/test/functional/feature_config_args.py#L287), even though we pass a realtive conf argument in.

This means that we are only testing the `os.path.join` sum of default `DataDir` and the relative arg.

Reply:
Closing as I think this is resolved